src/componententity.cpp

```diff
@@ -257,4 +257,16 @@ bool ComponentEntity::replaceComponent(const std::string &name, const ComponentP
     return status;
 }
 
+bool ComponentEntity::replaceComponent(const ComponentPtr &c1, const ComponentPtr &c2, bool searchEncapsulated)
+{
+    bool status = replaceComponent(mPimpl->findComponent(c1) - mPimpl->mComponents.begin(), c2);
+    if (searchEncapsulated && !status) {
+        for (size_t i = 0; i < componentCount() && !status; ++i) {
+            status = getComponent(i)->replaceComponent(c1, c2, searchEncapsulated);
+        }
+    }
+
+    return status;
+}
```

Comment:
Why reinvent the wheel (and potentially make it error prone when it comes to maintenance)? Why not simply:

```
bool ComponentEntity::replaceComponent(const ComponentPtr &c1, const ComponentPtr &c2, bool searchEncapsulated)
{
    return replaceComponent(c1->getName(), c2, searchEncapsulated);
}
```

Then again, this implies that there are no two components with the same name... and I can see that `removeComponent()` also duplicates code... In the end, I am happy with finding a component by pointer (as opposed to by name), but I am really not keen on duplicated code. We really ought to refactor the code to avoid all duplicates.

Reply:
Not sure how you'd do that here! We could have a find(componentptr) and a find(name) and then template the replaceComponent method I suppose?

Reply:
Would be good to be able to replace unnamed components without having to iterate over a list in user code, but that can always come later if requested.